Content.Server/SS220/SpiderQueen/Systems/SpiderQueenSystem.cs

```diff
@@ -0,0 +1,240 @@
+
+        if (entity.Comp.CocoonsCountToAnnouncement is { } value &&
+            entity.Comp.CocoonsList.Count >= value)
+            DoStationAnnouncement(entity);
+    }
+
+    /// <summary>
+    /// Do a station announcement if all conditions are met
+    /// </summary>
+    private void DoStationAnnouncement(EntityUid uid, SpiderQueenComponent? component = null)
+    {
+        if (!Resolve(uid, ref component) ||
+            component.IsAnnounced ||
+            !TryComp<TransformComponent>(uid, out var xform))
+            return;
+
+        var msg = Loc.GetString("spider-queen-warning",
+            ("location", FormattedMessage.RemoveMarkupOrThrow(_navMap.GetNearestBeaconString((uid, xform)))));
+        _chat.DispatchGlobalAnnouncement(msg, playSound: false, colorOverride: Color.Red);
+        _audio.PlayGlobal("/Audio/Misc/notice1.ogg", Filter.Broadcast(), true);
+        component.IsAnnounced = true;
+    }
+
+    /// <summary>
+    /// Converts hunger into blood points based on the <see cref="SpiderQueenComponent.HungerConvertCoefficient"/>
+    /// </summary>
+    private void ConvertHungerIntoBloodPoints(EntityUid uid, SpiderQueenComponent component, float amount, HungerComponent? hunger = null)
+    {
+        if (!Resolve(uid, ref hunger))
+            return;
+
+        var amountToMax = component.MaxBloodPoints - component.CurrentBloodPoints;
+        if (amountToMax <= FixedPoint2.Zero)
+            return;
+
+        var value = amount * component.HungerConvertCoefficient;
+        value = MathF.Min(value, (float)amountToMax);
+
+        var hungerDecreaseValue = -(value / component.HungerConvertCoefficient);
+        _hunger.ModifyHunger(uid, hungerDecreaseValue, hunger);
+        component.CurrentBloodPoints += value;
+        Dirty(uid, component);
+        UpdateAlert((uid, component));
+    }
+
+    private bool TryStartSpiderSpawnDoAfter(EntityUid spider,
+        TimeSpan doAfter,
+        EntityCoordinates coordinates,
+        List<EntitySpawnEntry> prototypes,
+        Vector2 offset,
+        FixedPoint2 cost)
+    {
+        var netCoordinates = GetNetCoordinates(coordinates);
+        var doAfterArgs = new DoAfterArgs(
+            EntityManager,
+            spider,
+            doAfter,
+            new SpiderSpawnDoAfterEvent()
+            {
+                TargetCoordinates = netCoordinates,
+                Prototypes = prototypes,
+                Offset = offset,
+                Cost = cost,
+            },
+            spider
+        )
+        {
+            Broadcast = false,
+            BreakOnDamage = false,
+            BreakOnMove = true,
+            NeedHand = false,
+            BlockDuplicate = true,
+            CancelDuplicate = true,
+            DuplicateCondition = DuplicateConditions.SameEvent
+        };
+
+        var started = _doAfter.TryStartDoAfter(doAfterArgs);
+        if (started)
+            return true;
+        else
+            return false;
```

Comment:
```suggestion
        return started;
```
т. к. это булиновая переменная